client/components/happychat/button.jsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import page from 'page';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import viewport from 'lib/viewport';
+import { openChat } from 'state/ui/happychat/actions';
+import Button from 'components/button';
+import Gridicon from 'components/gridicon';
+
+const HappychatButton = ( { translate, onOpenChat } ) =>
+	<Button compact borderless onClick={ onOpenChat }>
+		<Gridicon icon="comment" /> { translate( 'Support Chat' ) }
+	</Button>;
+
+const mapStateToProps = () => ( {} );
+
+const mapDispatchToProps = dispatch => {
+	return {
+		onOpenChat() {
+			if ( viewport.isMobile() ) {
+				// For mobile clients, happychat will always use the page compoent instead of the sidebar
+				page( '/me/chat' );
```

Comment:
Since this isn't related to dispatching actions, I'm not sure this is the best place for it. Instead, I'd think this would be a decision made by the component before choosing whether to dispatch. That way also dramatically simplifies the `mapDispatchToProps` to:

```js
export default connect( null, { openChat } )( localize( HappychatButton ) );
```

You'd need a click handler in your component, so I don't think a stateless function component will work (at least not without violating [`jsx-no-bind`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md) in spirit.